embassy-rp/src/uart/mod.rs

```diff
@@ -243,19 +308,95 @@ impl<'d, T: Instance> UartRx<'d, T, Blocking> {
     }
 }
 
+static UART_RX_WAKERS: [AtomicWaker; 2] = [AtomicWaker::new(), AtomicWaker::new()];
```

Comment:
could you do this with the "State" struct pattern? This is what the buffered uart does. and most other drivers do. 

Probably rename the buffered uart `state` to `buffered_state.`